quest/workflow.py

```diff
@@ -11,24 +13,48 @@
 WORKFLOW_RESULT = "WORKFLOW_RESULT"
 
 
+class WorkflowNotFoundException(Exception):
+    pass
+
+
+def find_frame():
+    outer_frame = inspect.currentframe()
+    is_workflow = False
+    while not is_workflow:
+        outer_frame = outer_frame.f_back
+        if outer_frame is None:
+            raise WorkflowNotFoundException("Workflow object not found in event stack")
+        try:
+            is_workflow = isinstance(outer_frame.f_locals['self'], Workflow)
+        except KeyError as e:
+            logging.debug(str(e) + ": outer_frame not called from a class")
```

Comment:
`is_workflow = isinstance(outer_frame.f_locals.get('self'), Workflow)`

No try-except necessary

Reply:
It actually is because if the outer frame doesn't have 'self' meaning it isn't a class it crashes. For example, what if the event is called from a function outside of a class after the workflow calls it?